src/main/scala/workflow/Rule.scala

```diff
@@ -0,0 +1,14 @@
+package workflow
+
+/**
+ * Represents a pipeline rule
```

Comment:
Transforms a Pipeline to a functionally-equivalent, different Pipeline. will extend the comments.